modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/RichText/RichText.es.js

```diff
@@ -62,32 +59,20 @@ const RichText = ({
 		>
 			<ClassicEditor
 				contents={currentValue}
-				data={currentValue}
 				editorConfig={editorConfig}
 				name={name}
 				onChange={(data) => {
 					if (currentValue?.trim() !== data?.trim()) {
-						setCurrentValue(data);
 						setDirty(true);
-
-						onChange({}, data);
+						onChange({target: {value: data}});
 					}
 					else if (!dirty) {
 						CKEDITOR.instances[name]?.resetUndo();
 					}
 				}}
-				onMode={({editor}) => {
-					if (editor.mode === 'source') {
-						editor.on('afterSetData', ({data}) => {
-							const {dataValue} = data;
-
-							setCurrentValue(dataValue);
-
-							onChange({}, dataValue);
-						});
-					}
-					else {
-						editor.removeListener('afterSetData');
+				onSetData={(event) => {
+					if (event.editor.mode === 'source') {
+						onChange({target: {value: event.data.dataValue}});
```

Comment:
It is completely optional, but you can take advantage of the [destructuring assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment)
```suggestion
				onSetData={({data: {dataValue: value}, editor: {mode}}) => {
					if (mode === 'source') {
						onChange({target: {value}});
```